lib/config/validation.spec.ts

```diff
@@ -912,5 +931,124 @@ describe('config/validation', () => {
         },
       ]);
     });
+
+    describe('globalOptions()', () => {
+      it('validates options of string type', async () => {
+        const config = {
+          binarySource: 'docker',
+          gitUrl: false as never,
```

Comment:
```suggestion
          // @ts-expect-error
          gitUrl: false,
```
use ts-expect-error

we should add this to the dev guide